src/consumer/component/sources/data-to-persist.ts

```diff
@@ -71,21 +71,29 @@ export default class DataToPersist {
     await this._persistFilesToFS();
     await this._persistSymlinksToFS();
   }
-  async persistAllToCapsule(capsule: Capsule) {
+  async persistAllToCapsule(capsule: Capsule, opts = { keepExistingCapsule: false }) {
     this._log();
     this._validateRelative();
-    await Promise.all(this.remove.map(pathToRemove => capsule.removePath(pathToRemove.path)));
-    await Promise.all(this.files.map(file => this._writeFileToCapsule(capsule, file)));
+    if (!opts.keepExistingCapsule) {
+      await Promise.all(this.remove.map(pathToRemove => capsule.removePath(pathToRemove.path)));
+    }
+    await Promise.all(
+      this.files.map(file => this._writeFileToCapsule(capsule, file, { overwriteExisting: !!opts.keepExistingCapsule }))
+    );
     await Promise.all(this.symlinks.map(symlink => this.atomicSymlink(capsule, symlink)));
   }
-  async _writeFileToCapsule(capsule: Capsule, file: AbstractVinyl) {
-    if (file.override === false) {
+  async _writeFileToCapsule(capsule: Capsule, file: AbstractVinyl, opts = { overwriteExisting: false }) {
+    if (file.override === false || opts.overwriteExisting) {
```

Comment:
I don't like the flow here, it's very confusing:
1. what is the different between `file.override` and `opts.overwriteExisting`? where do I get each one of them? who is stronger?
2. the condition is very weird, it's sounds something like - if (not override or override get here). like if (false or true) enter.
(I don't have in mind a better way, but let's try thinking about something more clear)